src/test/java/org/opensearch/knn/index/FaissIT.java

```diff
@@ -280,4 +293,106 @@ public void testEndToEnd_fromModel() throws IOException, InterruptedException {
             assertEquals(numDocs - i - 1, Integer.parseInt(results.get(i).getDocId()));
         }
     }
+
+    @SneakyThrows
+    public void testQueryWithFilter_withDifferentCombination_thenSuccess() {
```

Comment:
Should we add different tests for exact and ann searchers? I on a fence if we need it in IT as we already have unit test for this and it might be too low level for IT.

Reply:
No we should have integration tests. Even when we have Unit tests. The reason why I am combining both is, the setup for the tests are same. Hence I combined them at 1 place. Given that we already have so many integration tests and setting up each integration tests add more time. Hence, using a single setup.